server/src/main/java/com/google/musicanalysis/api/youtube/YoutubeRequest.java

```diff
@@ -0,0 +1,77 @@
+package com.google.musicanalysis.api.youtube;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class YoutubeRequest {
+  private static final String BASE_URL = "https://www.googleapis.com/youtube/v3/";
+  private static final String apiKey = "&key=";
+
+  private String operation;
+  private ArrayList<String> parameters = new ArrayList<String>();
+
+  public YoutubeRequest(String operation, String... parameters) {
+    this.operation = operation;
+    for (String parameter : parameters) {
+      this.parameters.add(parameter);
+    }
+  }
+
+  public String getResult() throws MalformedURLException, IOException {
+
+    StringBuffer buffer = new StringBuffer();
+
+    URL url = buildUrl();
+
+    BufferedReader in = new BufferedReader(new InputStreamReader(url.openStream(), "UTF-8"));
+
+    // reading chunks of response into the buffer
+    String chunk;
+    while ((chunk = in.readLine()) != null) {
+      buffer.append(chunk);
+    }
+
+    in.close();
+
+    return buffer.toString();
+  }
+
+  private URL buildUrl() throws MalformedURLException {
+    ArrayList<String> searchOps = new ArrayList<String>(Arrays.asList("?q=", "&videoCaption="));
+    ArrayList<String> commentOps = new ArrayList<String>(Arrays.asList("?part=", "&videoId="));
+    String parameterString;
+
+    if (this.operation == "search") {
+      parameterString = buildParams(searchOps);
+    } else {
+      parameterString = buildParams(commentOps);
+    }
+
+    String urlString = BASE_URL + this.operation + parameterString + apiKey;
+    URL url = new URL(urlString);
+    return url;
+  }
+
+  private String buildParams(ArrayList<String> ops) {
+    // More cases can be added in the future to this
+    switch (this.parameters.size()) {
```

Comment:
Instead of adding a new cases, this would scale better as a loop that just adds onto an existing parameter string for each item in the ArrayList.

Reply:
Ended up getting rid of this fn